PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+        var resultDict: [String: AnyObject] = [:]
+        let isInternalServerError = statusCode >= HTTPStatusCodes.internalServerError.rawValue
+        let isNotFoundError = statusCode == HTTPStatusCodes.notFoundError.rawValue
+
+        let successfullySynced = !(isInternalServerError || isNotFoundError)
+        resultDict[Backend.RCSuccessfullySyncedKey as String] = NSNumber(value: successfullySynced)
+
+        let hasAttributesResponseContainerKey = (response[Backend.RCAttributeErrorsResponseKey] != nil)
+        let attributesResponseDict = hasAttributesResponseContainerKey
+            ? response[Backend.RCAttributeErrorsResponseKey]
+            : response
+
+        if let attributesResponseDict = attributesResponseDict as? [String: Any] {
+            let hasAttributeErrors = (attributesResponseDict[Backend.RCAttributeErrorsKey] != nil)
+            if hasAttributeErrors {
+                resultDict[Backend.RCAttributeErrorsKey] = attributesResponseDict[Backend.RCAttributeErrorsKey] as AnyObject
+            }
+        }
+        return resultDict
+    }
+
+    private func handleSubscribedAttributesResult(statusCode: Int,
+                                                  response: [String: Any]?,
+                                                  error: Error?,
+                                                  completion: ((Error?) -> Void)?) {
+        guard let completion = completion else {
+            return
+        }
+
+        if let error = error {
+            completion(ErrorUtils.networkError(withUnderlyingError: error))
+            return
+        }
+
+        let responseError: Error?
+
+        if let response = response, statusCode > HTTPStatusCodes.redirect.rawValue {
+            let extraUserInfo = attributesUserInfoFromResponse(response: response, statusCode: statusCode)
+            responseError = ErrorUtils.backendError(withBackendCode: maybeNumberFromError(code: response["code"]),
+                                                    backendMessage: response["message"] as? String,
+                                                    extraUserInfo: extraUserInfo as [NSError.UserInfoKey: Any])
+        } else {
+            responseError = nil
+        }
+
+        completion(responseError)
+
+    }
+
+    private func handle(response: [String: Any]?, statusCode: Int, error: Error?, completion: ((Error?) -> Void)?) {
+        if let error = error {
+            completion?(ErrorUtils.networkError(withUnderlyingError: error))
+            return
+        }
+
+        if statusCode > HTTPStatusCodes.redirect.rawValue {
+            let code = maybeNumberFromError(code: response?["code"])
+            let message = response?["message"] as? String
+            let responseError = ErrorUtils.backendError(withBackendCode: code, backendMessage: message)
+            completion?(responseError)
+            return
```

Comment:
this `return` is useless, doesn't it?

Reply:
So, yes, currently, it's useless 😄  I put it in because there is always a risk that the next engineer could add more code below this so it makes people (probably me) think a little more.
That being said, I'm going to refactor this a tiny bit and use a `guard` because I think that will convey the intent better.